tests/test_simple.py

```diff
@@ -105,6 +105,63 @@ def test_relative_url(self):
         assert index["django-node"] == "django-node/"
 
 
+class TestArchiveLink:
+
+    """Tests for mousebender.simple.ArchiveLink."""
+
+    @pytest.mark.parametrize(
+        "archive_link",
+        [
+            simple.ArchiveLink(filename="B", url="A/B"),
+            simple.ArchiveLink(
+                filename="B",
+                url="A/B",
+                requires_python=packaging.specifiers.SpecifierSet(">=3.6"),
+            ),
+            simple.ArchiveLink(
+                filename="B",
+                url="A/B",
+                hash_=("sha256", "ABCDEF"),
+            ),
+            simple.ArchiveLink(
+                filename="B",
+                url="A/B",
+                gpg_sig=True,
+            ),
+            simple.ArchiveLink(filename="B", url="A/B", yanked=""),
+            simple.ArchiveLink(filename="B", url="A/B", yanked="oops!"),
+            simple.ArchiveLink(filename="B", url="A/B", metadata=("", "")),
+            simple.ArchiveLink(filename="B", url="A/B", metadata=("sha256", "ABCDEF")),
+            simple.ArchiveLink(
+                filename="B",
+                url="A/B",
+                requires_python=packaging.specifiers.SpecifierSet(">=3.6"),
+                hash_=("sha256", "ABCDEF"),
+                gpg_sig=True,
+                yanked="oops!",
+                metadata=("sha512", "GHIJKL"),
+            ),
+        ],
+    )
+    def test_str(self, archive_link):
```

Comment:
I like how this worked out as a test. Shows the user exactly how the intended  use case works, simply.